ast/src/parsed/mod.rs

```diff
@@ -46,6 +48,7 @@ impl SymbolCategory {
                 // Type constructors can also satisfy requests for values.
                 request == SymbolCategory::TypeConstructor || request == SymbolCategory::Value
             }
+            SymbolCategory::TraitDeclaration => request == SymbolCategory::Type,
```

Comment:
I think we need a new category called `Trait`. Let's consider `Into<A, B>` as a trait.

We cannot declare variables of that type. If you e.g. want a function parameter of that type you need
```
let<A, B, Into<A, B>> f: A -> B = |a| Into::into(a);
```

So you can "use" `Into` in an expression, but only as a path to get to one of the functions (`Into::into`).

The only point where you can use a trait (apart from an impl for that trait) is in the list of trait bounds:
`<A, B, Into<A, B>>`.

So I think it is fair to create a new category for that.

Reply:
Ah OK, sorry, you do have a new symbol category, but I think it should only be compatible with itself.